src/services/embeddings_service.py

```
@@ -0,0 +1,278 @@
+
+    def _initialize_client(self):
+        """Initialize the embedding client based on provider."""
+        if self.config.provider == "openai":
+            self._initialize_openai_client()
+        else:
+            raise ValueError(f"Unsupported embedding provider: {self.config.provider}")
+
+    def _initialize_openai_client(self):
+        """Initialize OpenAI client for embeddings."""
+        try:
+            import openai
```

Pylint (3.3.7)

[convention] Import outside toplevel (openai)

(C0415)